### PR TITLE
ptref: manage type.has_code(key, value) in filter

### PR DESCRIPTION
**Pull request by TeXitoi**
fix http://jira.canaltp.fr/browse/NAVP-260

### CODE REVIEW

File: source/jormungandr/tests/ptref_tests.py [hunk 1]
```diff
@@ -338,6 +338,13 @@ def test_vj_period_filter(self):
         assert code == 404
         assert get_not_null(response, 'error')['message'] == 'ptref : Filters: Unable to find object'
 
+    def test_line_by_code(self):
+        """test the filter=type.has_code(key, value)"""
+        response = self.query_region("lines?filter=line.has_code(codeB, B)&show_codes=true")
+        lines = get_not_null(response, 'lines')
+        assert len(lines) == 1
+        assert 'B' in [code['value'] for code in lines[0]['codes'] if code['type'] == 'codeB']
```
Comment (xlqian) [inline source/jormungandr/tests/ptref_tests.py]: use `filter()`

File: source/ptreferential/ptreferential.h [hunk 2]
```diff
@@ -64,11 +64,12 @@ struct Filter {
     Operator_e op; // comparison operator ("=")
     std::string value; // right value compared ("kikoolol") or arg used in method ("john")
     std::string method; // method called (has_headsign)
+    std::vector<std::string> args; // method arguments
 
     Filter(std::string object, std::string attribute, Operator_e op, std::string value) : object(object), attribute(attribute), op(op), value(value) {}
     Filter(std::string object, std::string value) : object(object), op(HAVING), value(value) {}
     Filter(std::string value) : object("journey_pattern_point"), op(AFTER), value(value) {}
-    Filter(std::string object, std::string method, std::string arg) : object(object), value(arg), method(method) {}
+    Filter(std::string object, std::string method, std::vector<std::string> args) : object(object), method(method), args(args) {}
```
Comment (xlqian) [inline source/ptreferential/ptreferential.h]: const &, at least on vector<string>?

Comment (TeXitoi) [inline source/ptreferential/ptreferential.h]: It will be moved.

**Comment by xlqian** [comment]
ok for me

**Comment by xlqian** [comment]
ok for me :+1:

**Pull request merged**